internal/resources/ekscluster/ekscluster_flatten_test.go

```diff
@@ -15,20 +15,170 @@ import (
 
 func TestFlattenCluterSpec(t *testing.T) {
 	tests := []struct {
-		description string
-		getInput    func() (*eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec, []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition)
-		expected    []interface{}
+		description  string
+		getInput     func(addonsConfig *eksmodel.VmwareTanzuManageV1alpha1EksclusterAddonsConfig) (*eksmodel.VmwareTanzuManageV1alpha1EksclusterSpec, []*eksmodel.VmwareTanzuManageV1alpha1EksclusterNodepoolDefinition)
+		addonsConfig *eksmodel.VmwareTanzuManageV1alpha1EksclusterAddonsConfig
```

Comment:
`getInput` is a generic function which can be used to add (or remove) `addonsConfig`. We don't need another field to specify the `addonsConfig` here.

Reply:
@asutoshpalai  I have made changes as requested